apps/oxlint/src/output_formatter/junit.rs

```diff
@@ -33,18 +33,21 @@ impl DiagnosticReporter for JUnitReporter {
 
 fn format_junit(diagnostics: &[Error]) -> String {
     let mut grouped: FxHashMap<String, Vec<&Error>> = FxHashMap::default();
-    let mut total_errors = 0;
-    let mut total_warnings = 0;
 
     for diagnostic in diagnostics {
         let info = Info::new(diagnostic);
         grouped.entry(info.filename).or_default().push(diagnostic);
     }
 
-    let mut test_suite = String::new();
-    for diagnostics in grouped.values() {
-        let diagnostic = diagnostics[0];
-        let filename = Info::new(diagnostic).filename;
+    let mut filenames: Vec<_> = grouped.keys().cloned().collect();
+    filenames.sort();
+
+    let mut total_errors = 0;
+    let mut total_warnings = 0;
+    let mut test_suites = Vec::new();
+
+    for filename in filenames {
+        let diagnostics = grouped.get(&filename).expect("filename collected from map");
```

Comment:
we generally have guarantees on presence here, but alternatively we we can also handle gracefully and continue 👁️